crates/noirc_evaluator/src/ssa/acir_gen/constraints.rs

```diff
@@ -430,25 +435,60 @@ pub(crate) fn to_radix(
     let shift = FieldElement::from(radix as i128);
     let mut result = Vec::new();
     let bit_size = bit_size_u32(radix);
-    for _ in 0..num_limbs {
-        let limb_witness = evaluator.add_witness_to_cs();
-        result.push(limb_witness);
-        let limb_expr = expression_from_witness(limb_witness);
-        digits = add(&digits, radix_pow, &limb_expr);
-        radix_pow = radix_pow.mul(shift);
-
-        if 1_u128 << (bit_size - 1) != radix as u128 {
-            try_range_constraint(limb_witness, bit_size, evaluator);
+    if endianness == Endian::Little {
+        for _ in 0..num_limbs {
+            let limb_witness = evaluator.add_witness_to_cs();
+            result.push(limb_witness);
+            let limb_expr = expression_from_witness(limb_witness);
+            digits = add(&digits, radix_pow, &limb_expr);
+            radix_pow = radix_pow.mul(shift);
+
+            if 1_u128 << (bit_size - 1) != radix as u128 {
+                try_range_constraint(limb_witness, bit_size, evaluator);
+            }
+            bound_constraint_with_offset(
+                &expression_from_witness(limb_witness),
+                &Expression::from_field(shift),
+                &Expression::one(),
+                bit_size,
+                evaluator,
+            );
+        }
+    } else {
+        // This is probably not a good solution.
```

Comment:
I don't understand, why don't you simply rename to_radix as to_radix_little and add to_radix which takes endianness as input, call to_radix_little and then reverse the result if big endian?

Reply:
I am not 100% sure but I guess you can call to_radix without any change (so little endian I guess), and then you handle the endianness in to_radix_base() by simply reversing the witness (or it this not working? I did not check carefully).
In any cases, I think it is best to not modify to_radix nor the to_radix directive (which will always be little endian) and abstract the endianess in acir-gen only.

Reply:
I don't understand why do you need the to_radix() function?
In case of big endian, to_radix_base() can simply call (result, bytes)= to_radix_little() and return the reversed result if big endian? can't it?
You should also remove the is_little_endian from the ToRadix directive (and give it the non-reversed result)

Reply:
because to_radix_base uses to_radix_little, it will always do little_endian in the constraints, so the directive does not need to support the big_endian case.

Reply:
When I said that we should remove to_radix, I did not mean to move its code into to_radix_base. You are modifying the 'bytes' expression instead of reversing the witness.

Reply:
yes